stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/plugin/listeners/HideChannelListener.kt

```diff
@@ -0,0 +1,41 @@
+package io.getstream.chat.android.client.experimental.plugin.listeners
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+
+@ExperimentalStreamChatApi
+/**
+ * Listener of [ChatClient.hideChannel] requests.
+ */
+public interface HideChannelListener {
+    /**
+     * Runs side effect before the request is launched.
+     *
+     * @param channelType Type of the requested channel.
+     * @param channelId Id of the requested channel.
+     * @param clearHistory Boolean, if you want to clear the history of this channel or not.
+     */
+    public suspend fun onHideChannelRequest(
+        channelType: String,
+        channelId: String,
+        clearHistory: Boolean
+    ) {
```

Comment:
Let's avoid those empty methods and keep only the signature. It is expected that users implement those methods, if they choose extend this interface

Reply:
Oh Yes,  you are right. I wrote it earlier than we merged the new `relevantPlugins` thing.